lib/tree.rb

```diff
@@ -16,40 +16,101 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(log n)
+  # Space Complexity: O(1)
+  def add_recursive(node, key, value)
```

Comment:
👍 , However the space complexity is O(log n) if the tree is balanced because of the call stack.